Implemented collection of resource stats for Windows Azure SQL Database.

martinknafve opened the pull request:
A couple of months ago a new dynamic management view was added to Windows Azure SQL Database. This view (resource_stats) contains some central performance metrics for the databases. For example, it allows you to see how many CPU cores you currently use. 

If you purchase Windows Azure SQL Database Premium Edition, you get reserved resource (for example 1 CPU core) and this commit allows you to see how much of these resources you actually used. We've been recommended by Microsoft to continously monitor the resource_stats, so I thought that adding it to the New Relic plug-in would be a good idea. We are currently running these changes in preview in our own production.

martinknafve committed:
Implemented collection of resource stats for Windows Azure SQL Database.

lars2893 commented:
Hey Martin,

First and foremost, thanks for your contribution!  We are going to hold off on a merge for this right now because that  view is in preview state, so the feature is subject to change.  If monitoring this data would be beneficial to you right now, I'd be happy to help you set up a private plugin so you can view and report on this data exactly as you would if it were merged.  Please don't hesitate to reach out if you have any questions or concerns!

Regards,

Adam

martinknafve commented:
Hey

That makes complete sense. We tend to use a lot of preview stuff in Windows Azure and I had actually forgotten that this view was still in preview. Of course you should postpone merging until it's production-ready (but when it is, I think many SQL Azure users will appreciate it).

Thanks for your suggestion on private plugin, but for now I've just changed the plugin guid locally and added the metrics to a dashboard so I already have it up and running. :-)

Have a nice weekend,

Regards
Martin

martinknafve committed:
Updated ToString() to include all fields, for completeness sake and to be consistent with other query types.

mharen commented:
:+1: This would be very helpful, especially the read/write IOP counters.

jstenhouse commented:
Thanks @martinknafve for the PR and sorry for the long delay! It looks like there is a conflict with your PR and can't be merged automatically. Can you rebase off of current master? 

Also, I saw that several of the columns in the resource_stats query are currently deprecated: http://msdn.microsoft.com/en-us/library/dn269979.aspx - Since they are going to be removed in a future release, can you only query the non-deprecated columns? Thanks!

Plasma commented:
Bump! Any chance this can be merged?

claassistantio commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/newrelic/newrelic_microsoft_sqlserver_plugin?pullRequest=13) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/newrelic/newrelic_microsoft_sqlserver_plugin?pullRequest=13) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/newrelic/newrelic_microsoft_sqlserver_plugin?pullRequest=13) it.</sub>

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/newrelic/newrelic_microsoft_sqlserver_plugin?pullRequest=13) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/newrelic/newrelic_microsoft_sqlserver_plugin?pullRequest=13) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/newrelic/newrelic_microsoft_sqlserver_plugin?pullRequest=13) it.</sub>
